alpa/device_mesh.py

```diff
@@ -521,7 +606,11 @@ def run_resharding_recv_task(self, uuid, buf_uuids, set_empty_buffer=True):
                                          recv_spec.shape, recv_spec.dtype)
             for recv_tile_spec in recv_spec.tile_specs:
                 recv_tile_spec: ReshardingTileSpec
-                self.recv_tile(buf_uuid, recv_spec.device_id,
+                if global_config.nccl_mode == "from_xla_extension":
+                    recv_tile_func = self.xla_nccl_recv_tile
+                else:
+                    recv_tile_func = self.recv_tile
```

Comment:
Try to merge these if/else statements. For example, move and merge them all into `__init__`

```python3
if global_config.nccl_mode == "from_xla_extension":
    self.recv_tile_func = self.xla_nccl_recv_tile
    self.send_tile_func = self.xla_nccl_send_tile
else:
    self.recv_tile_func = self.cupy_nccl_recv_tile
    self.send_tile_func = self.cupy_nccl_send_tile
```

Reply:
OK, that looks better.